flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -523,12 +525,12 @@ public void stop(SessionContext session) {
 		}
 
 		// store the result with a unique id (the job id for now)
-		final String resultId = jobGraph.getJobID().toString();
```

Comment:
Is `JobID` here possibly different from the actual `JobID` on running? I'm thinking of introduce a `getJobID` method in `Pipeline` even. A Pipeline corresponds to a job and is able to hold a job id.

Reply:
And then on translated, we use that JobID as the JobID set to JobGraph.

Reply:
It used to be that it was the `JobID` of the actual job. But that's not possible anymore because a `Pipeline` doesn't have a `JobID` anymore because we only generate the `JobGraph` when handing the `Pipeline` to the `Executor`.

Still using a `JobId` here is a bit weird, I'm pushing a commit that uses `UUID` instead. What do you think?

Reply:
> Still using a `JobID` here is a bit weird

If a result is associated with a job, why not using `JobID` as key? I'm ok with `UUID` in this patch but I think such a `UUID` is less expressive than a `JobID` if we're able to ensure the `JobID` to be the one of the actual job.

What do you think we let `Pipeline` take an optional JobID i.e. `@Nullable JobID getJobID()`? As mentioned above

> A Pipeline corresponds to a job and is able to hold a job id.

Reply:
Except this comment my concerns are covered by your follow-ups @aljoscha , thanks for your working!

I don't run this patch with a handy test since never I use SQL client myself. I expect Timo will give a final look so become a bit lazy :P

Reply:
We can think about adding that to `Pipeline`, yes, but so far a `JobID` is only created when a `JobGraph` is created. I think there might also not always be a strict correlation between the output and a job, i.e. one job might have multiple sinks, in which case we cannot tie the UUID of the result to the `JobID`.

Reply:
OK. Make sense to defer the discussion.